src/components/QuranReader/SidebarNavigation/ScrollableSelection.tsx

```diff
@@ -21,20 +31,37 @@ const ScrollableSelection = ({ items, searchPlaceholder, renderItem, getHref, is
     }
   }, [searchQuery, filteredItems, isJuz]);
 
+  const [scroll, selectedItemRef] = useScrollToElement<HTMLDivElement>({
```

Comment:
The scrolling options should be taken out of the entire component and stored as a const since its reference will change on every re-render also since the 3 components are using the same options, I suggest we export a common const that the 3 components will use. (`ComboboxItems` seems to also use the same exact config so the 4 components :) ). Maybe you can add it inside `useScrollToElement` itself something like:

```
export const SCROLL_TO_NEAREST_BLOCK_OPTIONS = {
  block: 'nearest', // 'block' relates to vertical alignment. see: https://stackoverflow.com/a/48635751/1931451 for nearest.
} as ScrollIntoViewOptions;
```